### ISSUE TITLE
Interpreting predictions of a Multi-Class versus a Multi-label TARs Classification

### ISSUE THREAD

**Issue by Shea-Fyffe** [flairNLP/flair#2410]
A quick clarification on the interpretation of multi-class versus multi-label. I've read some of the original research [(Halder et al., 2020)](https://aclanthology.org/2020.coling-main.285), but I'm not confident in my interpretation. I'm familiar with the `NLI` approach to zero-shot; however, I've read that Flair uses a slightly different architecture [(Peirsman, 2021)](https://nlp.town/blog/zero-shot-classification/)--moreover, I want to make sure this translates to a few-shot task.

Let's say I train the model in a few-shot scenario. In this example I'm only using **2** different classes. *Skip to the end to see my questions*

### 1. Setup analysis & corpus
```
# Flair for few-shot learning
from flair.data import Corpus, Sentence
from flair.datasets import SentenceDataset, ClassificationCorpus
from flair.trainers import ModelTrainer
from flair.models.text_classification_model import TARSClassifier
"""
####
"""
### Helper function
def convert_to_sentence_data(sentences, labels, task_name):
    res = SentenceDataset([
      Sentence(s).add_label(task_name, labels[i]) for i, s in enumerate(sentences)
          ])
    return res

# 17 training sentences --10 initiation of structure --7 consideration
train_sent = ['I let group members know what is expected of them.', 'I am friendly and approachable.', 'I encourage the use of uniform procedures.', 'I do little things to make it pleasant to be a member of the group.', 'I try out my ideas in the group.', 'I put suggestions made by the group into operation.', 'I make my attitudes clear to the group.', 'I treat all group members as my equals.', 'I decide what shall be done and how it shall be done.', 'I give advance notice of changes.', 'I assign group members to particular tasks.', 'I make sure that my part in the group is understood by the group members.', 'I look out for the personal welfare of group members.', 'I schedule the work to be done.', 'I am willing to make changes.', 'I maintain definite standards of performance.', 'I ask that group members to follow standard rules and regulations.']

# each training example is assigned to just 1 of the two labels 
train_labels = ['initiation of structure.', 'consideration.', 'initiation of structure.', 'consideration.', 'initiation of structure.', 'consideration.', 'initiation of structure.', 'consideration.', 'initiation of structure.', 'consideration.', 'initiation of structure.', 'initiation of structure.', 'consideration.', 'initiation of structure.', 'consideration.', 'initiation of structure.', 'initiation of structure.']

# use helper to convert to SentenceDataSet
train_data = convert_to_sentence_data(train_sent, train_labels, "multi_class_behaviors")

# make corpus (without setting multi_label = True)
crpus = Corpus(train=train_data)
label_dict = crpus.make_label_dictionary()
```
**Note:** all of the training examples are positively-keyed so they represent higher levels of the label construct.

### 2. Train a TARS model
For fine-tuning, I'm *not* specifying multi-label
```
#initialize the text classifier and trainer with corpus 
tars = TARSClassifier(task_name = "multi_class_behaviors",
                          label_dictionary = label_dict)

trainer = ModelTrainer(tars, crpus)
    
# train model
trainer.train(base_path="./example/dir/",
                    learning_rate=0.02,
                    mini_batch_size=4,
                    max_epochs=20,
                    embeddings_storage_mode="gpu")

#load model
ft_tars = TARSClassifier.load('./example/dir/best-model.pt')
```

### 3. Format unlabeled test data
```
#load test data
test_data = ['i care more about the work than my employees', 'get requests to increase or decrease budgets on projects and to close out projects as they are finished', 'marketing sales',  'delegated tasks for closing']

test_sentences = [Sentence(s) for s in test_data]
```

### 4a. Predict multi-class labels
```
#predict multi-class probabilities
ft_tars.predict(test_sentences, multi_class_prob = True)
```

### 4b. Predict multi-label probabilities
```
# switch to multi-label task
ft_tars.add_and_switch_to_new_task(
                          task_name = "multi_label_behaviors",
                          label_dictionary = label_dict,
                          multi_label = True,
                          multi_label_threshold = 0.0)
#predict multi-label probabilities
ft_tars.predict(test_sentences)
```

***Questions:*** 

- When interpreting the prediction from section `4a` (i.e., multi-class) I should assume that these values represent the probability that the model thought a test sentence was *related* to `initiating structure` OR `consideration`?
- This is a tricky one (for me). When interpreting the prediction from section `4b` (i.e., multi-label) I should assume that values for `consideration`--for instance--represent the probability that a sentence is *related* to positive consideration? In other words, is this synonymous with entailment in an NLI task? Since my training set only included examples that represent positive consideration and not inconsideration/negative consideration, I must interpret it this way?
- Overall, do the predicted values represent just the strength of a relationship or do they also include direction? So a test case that depicts being *inconsiderate* (e.g., 'i care more about the work than my employees') is likely to receive a lower multi-label predicted value along with test cases that are unrelated to consideration (e.g.,  'marketing sales')?

In summary, my goal is to produce roughly 3 variables for each of the 2 labels (similar to the common `NLI` framework or a even a 3-label sentiment analysis); where one pair are probabilities of a positive/entailment relationship one pair are probabilities of a negative/contradiction relationship, and one pair neutral/unrelated probabilities.

This is somewhat related to: #2365

**Comment by Shea-Fyffe** [comment]
*bump*

**Comment by Shea-Fyffe** [comment]
Maybe another time ¯\_(ツ)_/¯
